cpp/src/arrow/compute/api_scalar.h

```diff
@@ -188,23 +218,21 @@ enum CompareOperator : int8_t {
   LESS_EQUAL,
 };
 
-struct CompareOptions : public FunctionOptions {
-  explicit CompareOptions(CompareOperator op) : op(op) {}
-
+class ARROW_EXPORT CompareOptions : public FunctionOptions {
+ public:
+  explicit CompareOptions(CompareOperator op);
```

Comment:
I see this is necessary; that's unfortunate. Could you open a follow up JIRA to remove the binding to `CompareOptions` and demote this to a deprecated struct?

Reply:
Filed ARROW-13219